content/applications/productivity/mail_plugins/gmail.rst

```diff
@@ -2,94 +2,162 @@
-      Make sure you are using the legacy editor; otherwise the *Deploy from manifest* functionality
-      may not be available.
+.. image:: gmail/authorize-access.png
+   :align: center
+   :alt: The Authorize Access button in the right sidebar of the Odoo plugin panel.
+
+Next, click :guilabel:`Login`. Then, enter the URL of the Odoo database that the user wishes to
+connect to the Gmail inbox, and log in to the database.
+
+.. note::
+   Use the general URL for the database, not the URL of a specific page in the database. For
+   example, use `https://mycompany.odoo.com`, not
+   `https://mycompany.odoo.com/web#cids=1&action=menu`.
+
+Finally, click :guilabel:`Allow` to let Gmail access the Odoo database. The browser will then show a
+:guilabel:`Success!` message. After that, close the window. The Gmail inbox and Odoo database are
+now connected.
+
+Odoo On-Premise users
+=====================
+
+For databases hosted on servers other than Odoo Online (or Odoo.sh), follow the steps below to
+configure the Gmail Plugin.
+
+.. note::
+   As part of their security guidelines, Google requires add-on creators to provide a list of URLs
+   that can be used in actions and redirections launched by the add-on. This protects users by
+   ensuring, for example, that no add-on redirects users toward a malicious website. (Read more on
+   `Google Apps Script <https://developers.google.com/apps-script/manifest/allowlist-url>`_.)
+   
+   Since Odoo can only list the `odoo.com` domain and not every on-premise customer’s unique server
+   domain, on-premise customers cannot install the Gmail Plugin from the Google Workspace
+   Marketplace.
+
+Install the Gmail Plugin
+------------------------
 
-      .. image:: gmail/legacy-editor.png
-         :align: center
-         :alt: Using the legacy editor in the Gmail Plugin Apps Script project
+First, access the `GitHub repository <https://github.com/odoo/mail-client-extensions>`_ for the
+Odoo Mail Plugins. Next, click on the green :guilabel:`Code` button. Then, click
+:guilabel:`Download ZIP` to download the Mail Plugin files onto the user's computer.
 
-#. Click on *Install add-on*. A "Deployment installed" notification should appear. You can then
-   click on *Close*.
+.. image:: gmail/gh-download-zip.png
+   :align: center
+   :alt: Download the ZIP file from the Odoo GitHub repository for Mail Plugins.
 
-   .. image:: gmail/install-add-on.png
-      :align: center
-      :alt: Installing the Gmail Plugin from the Apps Script project
+Open the ZIP file on the computer. Then, go to :menuselection:`mail-client-extensions-master -->
+gmail --> src --> views`, and open the :guilabel:`login` TS file using any text editor software,
+such as Notepad (Windows), TextEdit (Mac), or Visual Studio Code.
 
-.. _mail-plugin/gmail/configure-mailbox:
+Delete the following three lines of text from the :guilabel:`login` TS file:
 
-Configure your Gmail mailbox
-----------------------------
+.. code-block:: python
```

Comment:
Remove `python` and the CI error will resolve (this isn't actually python per se as it's in a `.ts` file not a `.py` file).
```suggestion
.. code-block::
```